server/pulp/plugins/conduits/repo_publish.py

```diff
@@ -64,11 +63,9 @@ def last_publish(self):
         :raises DistributorConduitException: if any errors occur
         """
         try:
-            collection = RepoDistributor.get_collection()
-            distributor = collection.find_one({'repo_id': self.repo_id, 'id': self.distributor_id})
-            if distributor is None:
-                raise pulp_exceptions.MissingResource(self.repo_id)
-            return dateutils.ensure_tz(distributor['last_publish'])
+            dist = model.Distributor.objects.get_or_404(repo_id=self.repo_id,
+                                                        distributor_id=self.distributor_id)
```

Comment:
I recommend adding ``.only('last_publish')``, if ``.get()`` supports that. Let me know if you know for sure whether it does or doesn't support that.